docs/code_gen_options.rst

```diff
@@ -145,6 +145,21 @@ This also means that, whenever the length of an array is accessed using '.length
 Note that this optimization does not cover every case. When an array is passed
 as a function argument, for instance, the length is unknown.
 
+``wasm-opt`` optimization passes
+--------------------------------
+
+For the Substrate target, optimization passes from the `binaryen <https://github.com/WebAssembly/binaryen>`_ ``wasm-opt`` 
+tool can be applied. This may vastly shrink the Wasm code size and make it more efficient.
```

Comment:
- wasm should be either lower case or upper case
- "may vastly" is a bit odd. Sometimes it only shrinks it a little bit. How about simply "may shrink"?
- "makes it more effcient" (make => makes)

Reply:
> * wasm should be either lower case or upper case

Ideally it should conform to how we've been writing `wasm` in our docs.

Reply:
The abbreviation is called [`Wasm`](https://webassembly.org/) as I wrote it

Reply:
> * "may vastly" is a bit odd. Sometimes it only shrinks it a little bit. How about simply "may shrink"?
> * "makes it more effcient" (make => makes)

@xermicus This has not been solved yet.